### PR TITLE
list-skus: --all should be a flag

### PR DESCRIPTION
**Pull request by yugangw-msft**
Guess no one is really using this flag, but since I found it when investigated the icm ticket of "https://icm.ad.msft.net/imp/v3/incidents/details/101603394/home", so I am fixing this

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yugangw-msft** [comment]
@adewaleo, please  take a look at my comment in the icm ticket for context

File: src/command_modules/azure-cli-vm/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Release History
 ===============
+* vm list-skus: --all should be a flag instead of an argument requiring values
```
Comment (tjprescott) [inline src/command_modules/azure-cli-vm/HISTORY.rst]: I would reword this. 

`* vm list-skus: Allow use of `--all` in place of `--all true`
